### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by calebchris000**
### In this project:
- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.

**Cheers.** 🥂

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @calebchris000, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no professional documentations, kindly make sure to use the [README template](https://github.com/microverseinc/readme-template) provided by microverse, also make sure the `.github` folder exist in your repo so the linters runs as expected

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Ismailco — COMMENTED** [review]
Hi @calebchris000, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no professional documentations, kindly make sure to use the [README template](https://github.com/microverseinc/readme-template) provided by microverse, also make sure the `.github` folder exist in your repo so the linters runs as expected

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Kamba56** [comment]
Hi @calebchris000

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly set up linters correctly

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230606140040_create_post.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.string :timestamp
+      t.integer :comment_counter
+      t.integer :likes_counter
+      t.references :users, null: false, foreign_key: true, index: true
```
Comment (DuaneDave) [inline db/migrate/20230606140040_create_post.rb]: To follow the ERD correctly, the foreign key is supposed to be an AuthorId. So recommend changing the user to author like this:

```
t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20230606175404_create_comment.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateComment < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.string :timestamps
+      t.references :users, null: false, foreign_key: true, index: true
```
Comment (DuaneDave) [inline db/migrate/20230606175404_create_comment.rb]: To follow the ERD correctly, the foreign key is supposed to be an AuthorId. So recommend changing the user to author like this:

```
`t.references :author, foreign_key: { to_table: 'users' }`
```

File: db/migrate/20230606140456_create_like.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateLike < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.string :timestamp
+      t.references :users, null: false, foreign_key: true, index: true
```
Comment (DuaneDave) [inline db/migrate/20230606140456_create_like.rb]: To follow the ERD correctly, the foreign key is supposed to be an AuthorId. So recommend changing the user to author like this:

```
`t.references :author, foreign_key: { to_table: 'users' }`
```

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @calebchris000 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Clean and well written migrations
✔️ Correct workflow in use

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230606140456_create_like.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ class CreateLike < ActiveRecord::Migration[7.0]
   def change
     create_table :likes do |t|
       t.string :timestamp
-      t.references :users, null: false, foreign_key: true, index: true
+      t.references :authors, null: false, foreign_key: true, index: true
```
Comment (afanimartin) [inline db/migrate/20230606140456_create_like.rb]: - Please change reference to: `t.references :author, null: false, foreign_key: { to_table: 'users' }, index: true`. Do the same for the other migration files. For `Post` and `Comment`.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @calebchris000, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Changes made as requested though more changes need to be made.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @calebchris000 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great job implementing all requested changes.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**